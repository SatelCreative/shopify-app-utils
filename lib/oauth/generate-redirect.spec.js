const generateRedirect = require('./generate-redirect');

describe('Generate Redirect', () => {
  it('should generate a url', () => {
    expect(
      generateRedirect({
        shop: 'example.myshopify.com',
        apiKey: 'key',
        scope: ['read_products'],
        nonce: 'nonce',
        redirect: 'https://example.com',
      }),
    ).toMatchSnapshot();
  });

  it('should generate an online url', () => {
    expect(
      generateRedirect({
        shop: 'example.myshopify.com',
        apiKey: 'key',
        scope: ['read_products'],
        nonce: 'nonce',
        redirect: 'https://example.com',
        online: true,
      }),
    ).toMatchSnapshot();
  });

  it('should generate an iframe redirect', () => {
    expect(
      generateRedirect({
        shop: 'example.myshopify.com',
        apiKey: 'key',
        scope: ['read_products'],
        nonce: 'nonce',
        redirect: 'https://example.com',
        iframe: true,
      }),
    ).toMatchSnapshot();
  });
});
