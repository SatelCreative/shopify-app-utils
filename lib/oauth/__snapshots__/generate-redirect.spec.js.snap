// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Redirect should generate a url 1`] = `"https://example.myshopify.com/admin/oauth/authorize?client_id=key&state=nonce&redirect_uri=https%3A%2F%2Fexample.com&scope=read_products"`;

exports[`Generate Redirect should generate an iframe redirect 1`] = `"<!DOCTYPE html><html><head><script type=\\"text/javascript\\">window.top.location.href = \\"https://example.myshopify.com/admin/oauth/authorize?client_id=key&state=nonce&redirect_uri=https://example.com&scope=read_products\\"</script></head></html>"`;

exports[`Generate Redirect should generate an online url 1`] = `"https://example.myshopify.com/admin/oauth/authorize?client_id=key&state=nonce&redirect_uri=https%3A%2F%2Fexample.com&scope=read_products&grant_options%5B%5D=per-user"`;
